src/configure/mod.rs

```diff
@@ -338,6 +341,11 @@ async fn configure(ctx: &Context, param: &mut LoginParam) -> Result<()> {
     drop(imap);
 
     progress!(ctx, 910);
+
+    // Make sure update_device_chats is done before we set the "configured" config
+    // so that no DC_EVENT_MSGS_CHANGED is sent (this would confuse bots and python tests)
```

Comment:
how is sending DC_EVENT_MSGS_CHANGED avoided by that?

sidenote: i think, in currently implementations these events are not needed, however, in general, it seems to be more consistent to emit them to make sure a maybe already displayed chatlist is updated.

Reply:
The idea was to be nicer to bots and tests. I think I'll just add a bot flag and fix the tests in another way.